packages/cart/src/services/cart-module.ts

```diff
@@ -176,10 +191,378 @@ export default class CartModuleService implements ICartModuleService {
+    @MedusaContext() sharedContext: Context = {}
+  ) {
+    const items = await this.lineItemService_.list(
+      filters,
+      config,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<CartTypes.CartLineItemDTO[]>(
+      items,
+      {
+        populate: true,
+      }
+    )
+  }
+
+  @InjectManager("baseRepository_")
+  async listShippingMethods(
+    filters = {},
+    config: FindConfig<CartTypes.CartShippingMethodDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ) {
+    const methods = await this.shippingMethodService_.list(
+      filters,
+      config,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<
+      CartTypes.CartShippingMethodDTO[]
+    >(methods, {
+      populate: true,
+    })
+  }
+
+  addLineItems(
+    data: CartTypes.AddLineItemsDTO,
+    sharedContext?: Context
+  ): Promise<CartTypes.CartLineItemDTO[]>
+  addLineItems(
+    data: CartTypes.AddLineItemsDTO[],
+    sharedContext?: Context
+  ): Promise<CartTypes.CartLineItemDTO[]>
+  addLineItems(
+    cartId: string,
+    items: CartTypes.CreateLineItemDTO[],
+    sharedContext?: Context
+  ): Promise<CartTypes.CartLineItemDTO[]>
+
+  @InjectManager("baseRepository_")
+  async addLineItems(
+    cartIdOrData:
+      | string
+      | CartTypes.AddLineItemsDTO[]
+      | CartTypes.AddLineItemsDTO,
+    dataOrSharedContext?: CartTypes.CreateLineItemDTO[] | Context,
+    sharedContext?: Context
+  ): Promise<CartTypes.CartLineItemDTO[]> {
+    if (isString(cartIdOrData)) {
+      return await this.addLineItems_(
+        cartIdOrData,
+        dataOrSharedContext as CartTypes.CreateLineItemDTO[],
+        sharedContext
+      )
+    }
+
+    if (Array.isArray(cartIdOrData)) {
+      return await this.addLineItemsBulk_(
+        cartIdOrData,
+        dataOrSharedContext as Context
+      )
+    }
+
+    return await this.addLineItemsBulk_(
+      [cartIdOrData],
+      dataOrSharedContext as Context
+    )
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  protected async addLineItems_(
+    cartId: string,
+    data: CartTypes.CreateLineItemDTO[],
+    sharedContext?: Context
+  ): Promise<CartTypes.CartLineItemDTO[]> {
+    return await this.addLineItemsBulk_(
+      [{ cart_id: cartId, items: data }],
+      sharedContext
+    )
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  protected async addLineItemsBulk_(
+    data: CartTypes.AddLineItemsDTO[],
+    sharedContext?: Context
+  ): Promise<CartTypes.CartLineItemDTO[]> {
+    const lineItemsMap = new Map<string, CartTypes.CreateLineItemDTO[]>()
+
+    for (const lineItemData of data) {
+      lineItemsMap.set(lineItemData.cart_id, lineItemData.items)
+    }
+
+    const items = await promiseAll(
+      [...lineItemsMap].map(async ([cartId, lineItems]) => {
+        return await this.lineItemService_.create(
+          cartId,
+          lineItems,
+          sharedContext
+        )
+      })
+    )
```

Comment:
question: should the internal line item service always take a collection of cart id - items? such that we can bulk all of that instead of looping? here mikro orm will handle it under the hood as part of the UoW but it is more to be future proof